### ISSUE TITLE
Cannot import imglyb in init method

### ISSUE THREAD

**Issue by jjahanip** [imagej/imagej.py#6]
I have successfully installed `pyjnius`, `imglib2-imglyb` and `imagej` and I am able to import `pyjnius` and `imglyb` without any problem.

The problem comes when I want to import `imglyb` in **init method**. I pass the correct path to `ij_dir` and 363 Jars are added to CLASSPATH. But, when it gets to the `import imglyb` line, it throws this error:

```
Traceback (most recent call last):
  File "E:/test_project/main.py", line 8, in <module>
    ij = imagej.init('D:/fiji-win64/Fiji.app')
  File "D:\Miniconda3\envs\test\lib\site-packages\imagej\imagej.py", line 204, in init
    import imglyb
  File "D:\Miniconda3\envs\test\lib\site-packages\imglyb\__init__.py", line 50, in <module>
    from .util import \
  File "D:\Miniconda3\envs\test\lib\site-packages\imglyb\util.py", line 11, in <module>
    from .imglib_ndarray import ImgLibReferenceGuard
  File "D:\Miniconda3\envs\test\lib\site-packages\imglyb\imglib_ndarray.py", line 9, in <module>
    Intervals = autoclass( 'net.imglib2.util.Intervals' )
  File "D:\Miniconda3\envs\test\lib\site-packages\jnius\reflect.py", line 158, in autoclass
    c = find_javaclass(clsname)
  File "jnius\jnius_export_func.pxi", line 25, in jnius.find_javaclass
jnius.JavaException: Class not found b'net/imglib2/util/Intervals'
```

and this is the exact code that I used:

```
import imagej
ij = imagej.init('D:/fiji-win64/Fiji.app')
```

I am on 64bit windows and I created conda environment (called test) for this code.

**Comment by hanslovsky** [comment]
Please
 - print a list of the jars that were found. That should contain a jar like `imglib2-<version>.jar`.
 - Print the contents of `'D:/fiji-win64/Fiji.app/jars'`

**Comment by jjahanip** [comment]
contents of `'D:\fiji-win64\Fiji.app\jars'`:
- ahocorasick-0.2.4.jar
- args4j-2.0.25.jar
- asm-4.0.jar
- asm-analysis-4.0.jar
- asm-commons-4.0.jar
- asm-tree-4.0.jar
- asm-util-4.0.jar
- autocomplete-2.6.1.jar
- base64-2.3.8.jar
- batik-anim-1.10.jar
- batik-awt-util-1.10.jar
- batik-bridge-1.10.jar
- batik-codec-1.9.jar
- batik-constants-1.10.jar
- batik-css-1.10.jar
- batik-dom-1.10.jar
- batik-ext-1.10.jar
- batik-gvt-1.10.jar
- batik-i18n-1.10.jar
- batik-parser-1.10.jar
- batik-script-1.10.jar
- batik-svg-dom-1.10.jar
- batik-svggen-1.10.jar
- batik-transcoder-1.9.jar
- batik-util-1.10.jar
- batik-xml-1.10.jar
- bigdataviewer-core-5.0.0.jar
- bigdataviewer-vistools-1.0.0-beta-11.jar
- bio-formats
- blas-0.8.jar
- bounce-0.18.jar
- bsh-2.0b6.jar
- bytelist-1.0.11.jar
- caffeine-2.4.0.jar
- clojure-1.8.0.jar
- codemodel-2.6.jar
- collections-generic-4.01.jar
- commons-beanutils-1.9.3.jar
- commons-codec-1.11.jar
- commons-collections-3.2.2.jar
- commons-compress-1.4.1.jar
- commons-io-2.6.jar
- commons-lang-2.6.jar
- commons-lang3-3.7.jar
- commons-logging-1.2.jar
- commons-math-2.2.jar
- commons-math3-3.6.1.jar
- commons-vfs2-2.0.jar
- datasets-0.8.1906.jar
- ejml-0.24.jar
- eventbus-1.4.jar
- f2jutil-0.8.jar
- ffmpeg-2.6.1-0.11.jar
- fiji-2.0.0-pre-6.jar
- fiji-compat-2.0.3.jar
- fiji-lib-2.1.2.jar
- filters-2.0.235.jar
- gcc-runtime-0.8.1906.jar
- gentyref-1.1.0.jar
- gluegen-rt-2.3.2.jar
- gluegen-rt-main-2.3.2.jar
- gpars-1.2.1.jar
- graphics-0.8.1906.jar
- grDevices-0.8.1906.jar
- groovy-2.4.8.jar
- gson-2.8.5.jar
- guava-21.0.jar
- ij-1.52c.jar
- ij1-patcher-0.12.9.jar
- image4j-0.7.jar
- imagej-2.0.0-rc-67.jar
- imagej-common-0.26.1.jar
- imagej-deprecated-0.1.2.jar
- imagej-launcher-4.0.5.jar
- imagej-legacy-0.30.0.jar
- imagej-mesh-0.7.0.jar
- imagej-notebook-0.2.3.jar
- imagej-ops-0.41.1.jar
- imagej-plugins-commands-0.8.0.jar
- imagej-plugins-tools-0.3.1.jar
- imagej-plugins-uploader-ssh-0.3.2.jar
- imagej-plugins-uploader-webdav-0.2.2.jar
- imagej-scripting-0.8.1.jar
- imagej-ui-awt-0.3.1.jar
- imagej-ui-swing-0.21.4.jar
- imagej-updater-0.9.2.jar
- imglib2-5.3.0.jar
- imglib2-algorithm-0.9.0.jar
- imglib2-algorithm-fft-0.2.0.jar
- imglib2-algorithm-gpl-0.2.3.jar
- imglib2-cache-1.0.0-beta-9.jar
- imglib2-ij-2.0.0-beta-42.jar
- imglib2-realtransform-2.0.0.jar
- imglib2-roi-0.5.1.jar
- imglib2-ui-2.0.0.jar
- invokebinder-1.2.jar
- itextpdf-5.5.13.jar
- ivy-2.4.0.jar
- j3dcore-1.6.0-scijava-2.jar
- j3dutils-1.6.0-scijava-2.jar
- jai-codec-1.1.3.jar
- jai-core-1.1.3.jar
- jama-1.0.3.jar
- javacpp-0.11.jar
- javassist-3.22.0-GA.jar
- jcodings-1.0.10.jar
- jcommander-1.48.jar
- jcommon-1.0.24.jar
- jdatepicker-1.3.2.jar
- jdom2-2.0.6.jar
- jep-2.4.2.jar
- jffi-1.2.7-native.jar
- jffi-1.2.7.jar
- jfreechart-1.5.0.jar
- jgoodies-common-1.7.0.jar
- jgoodies-forms-1.7.2.jar
- jgrapht-0.8.3.jar
- jgraphx-3.4.1.3.jar
- jhdf5-14.12.6.jar
- jhotdraw-7.6.0.jar
- jitk-tps-3.0.0.jar
- jna-4.5.1.jar
- jnr-constants-0.8.5.jar
- jnr-enxio-0.4.jar
- jnr-ffi-1.0.7.jar
- jnr-netdb-1.1.2.jar
- jnr-posix-3.0.1.jar
- jnr-unixsocket-0.3.jar
- jnr-x86asm-1.0.2.jar
- joal-2.3.2.jar
- joal-main-2.3.2.jar
- jocl-2.3.2.jar
- jocl-main-2.3.2.jar
- joda-time-2.9.9.jar
- jogl-all-2.3.2.jar
- jogl-all-main-2.3.2.jar
- joni-2.1.1.jar
- jpedalSTD-2.80b11.jar
- jply-0.2.0.jar
- jruby-core-1.7.12.jar
- jruby-stdlib-1.7.12.jar
- jsch-0.1.54.jar
- json-20090211.jar
- jsoup-1.7.2.jar
- jsr166y-1.7.0.jar
- jtransforms-2.4.jar
- jung-api-2.0.1.jar
- jung-graph-impl-2.0.1.jar
- JWlz-1.4.0.jar
- jython-shaded-2.7.1.jar
- jzlib-1.1.3.jar
- kryo-2.24.0.jar
- languagesupport-2.6.0.jar
- lapack-0.8.jar
- legacy-imglib1-1.1.8.jar
- log4j-1.2.17.jar
- logback-classic-1.2.3.jar
- logback-core-1.2.3.jar
- mapdb-1.0.3.jar
- markdownj-0.3.0-1.0.2b4.jar
- maven-scm-api-1.4.jar
- maven-scm-provider-svn-commons-1.4.jar
- maven-scm-provider-svnexe-1.4.jar
- metadata-extractor-2.6.2.jar
- methods-0.8.1906.jar
- miglayout-3.7.4-swing.jar
- mines-jtk-20151125.jar
- minimaven-2.2.1.jar
- minlog-1.2.jar
- mpicbg-1.2.0.jar
- mtj-1.0.4.jar
- multiverse-core-0.7.0.jar
- nailgun-server-0.9.1.jar
- native-lib-loader-2.2.0.jar
- netcdf-4.3.22.jar
- netlib-java-0.9.3-renjin-patched-2.jar
- object-inspector-0.1.jar
- objenesis-2.1.jar
- ojalgo-45.1.0.jar
- OMEVisual-1.1.1.jar
- op-finder-0.1.3.jar
- opencsv-3.9.jar
- options-1.1.jar
- pal-optimization-2.0.1.jar
- parsington-1.0.1.jar
- perf4j-0.9.13.jar
- plexus-utils-1.5.6.jar
- postgresql-9.1-901-1.jdbc4.jar
- prettytime-4.0.1.Final.jar
- regexp-1.3.jar
- renjin-appl-0.8.1906.jar
- renjin-core-0.8.1906.jar
- renjin-gnur-runtime-0.8.1906.jar
- renjin-script-engine-0.8.1906.jar
- rhino-1.7.6.jar
- rsyntaxtextarea-2.6.1.jar
- scala-compiler-2.12.1.jar
- scala-library-2.12.1.jar
- scala-reflect-2.12.1.jar
- scala-xml_2.12-1.0.6.jar
- scifio-0.37.0.jar
- scifio-bf-compat-2.0.3.jar
- scifio-cli-0.4.0.jar
- scifio-hdf5-0.1.2.jar
- scifio-jai-imageio-1.1.1.jar
- scifio-lifesci-0.7.1.jar
- scifio-ome-xml-0.14.3.jar
- scijava-common-2.74.1.jar
- scijava-config-2.0.2.jar
- scijava-plugins-commands-0.2.3.jar
- scijava-plugins-platforms-0.3.1.jar
- scijava-plugins-text-markdown-0.1.3.jar
- scijava-plugins-text-plain-0.1.3.jar
- scijava-search-0.4.1.jar
- scijava-ui-awt-0.1.6.jar
- scijava-ui-swing-0.10.0.jar
- script-editor-0.1.6.jar
- scripting-beanshell-0.3.3.jar
- scripting-clojure-0.1.6.jar
- scripting-groovy-0.2.7.jar
- scripting-java-0.4.1.jar
- scripting-javascript-0.4.4.jar
- scripting-jruby-0.2.5.jar
- scripting-jython-0.4.2.jar
- scripting-renjin-0.2.2.jar
- scripting-scala-0.2.1.jar
- slf4j-api-1.7.25.jar
- snakeyaml-1.13.jar
- spim_data-2.2.1.jar
- stats-0.8.1906.jar
- swing-checkbox-tree-1.0.2.jar
- swing-worker-1.1.jar
- swingx-1.6.1.jar
- T2-NIT-1.1.2.jar
- T2-TreelineGraph-1.1.2.jar
- trakem2-transform-1.0.1.jar
- trove4j-3.0.3.jar
- udunits-4.3.18.jar
- ui-behaviour-1.7.3.jar
- utils-0.8.1906.jar
- vecmath-1.6.0-scijava-2.jar
- VectorString-2.0.1.jar
- VIB-lib-2.2.0.jar
- weave_jy2java-2.1.0.jar
- weka-dev-3.9.2.jar
- win64
- xerbla-0.8.jar
- xercesImpl-2.8.1.jar
- xml-apis-ext-1.3.04.jar
- xmlgraphics-commons-2.3.jar
- xmpcore-5.1.2.jar
- xz-1.0.jar
- yecht-1.0.jar

Following `.jar`s where found:
- imglib2-5.3.0.jar
- imglib2-algorithm-0.9.0.jar
- imglib2-algorithm-fft-0.2.0.jar
- imglib2-algorithm-gpl-0.2.3.jar
- imglib2-cache-1.0.0-beta-9.jar
- imglib2-ij-2.0.0-beta-42.jar
- imglib2-realtransform-2.0.0.jar
- imglib2-roi-0.5.1.jar
- imglib2-ui-2.0.0.jar

**Comment by ctrueden** [comment]
@jjahanip Does your Fiji work normally when you start it as an application? Does the following Jython script work in the Script Editor?
```python
from net.imglib2.util import Intervals
print(Intervals)
```

@hanslovsky Do you think this could be a Windows-specific issue similar to [the one discussed on the forum](http://forum.imagej.net/t/analysis-with-imagej-and-visualization-in-the-jupyter-notebook/11052/22)?

Has anyone successfully gotten this working on Windows? Personally I never tried it.

**Comment by jjahanip** [comment]
My Fiji works normally when I start the application.

Running the Jython script prints out: `<type 'net.imglib2.util.Intervals'>`

**Comment by hanslovsky** [comment]
@ctrueden That could be related, indeed.
PyJNIus/imglyb from conda used to work on Windows but there are issues now:
http://forum.imagej.net/t/analysis-with-imagej-and-visualization-in-the-jupyter-notebook/11052/20?u=hanslovsky
When I try to re-package PyJNIus for conda I encounter the same error. I will need package maintainers for Windows (and maybe OSX) who can deal with these kinds of issues. I have no clue of Windows and can't do it.
http://forum.imagej.net/t/imglyb-and-pyjnius-conda-package-maintainers-needed-for-windows-and-osx/11420

**Comment by hanslovsky** [comment]
@jjahanip 
This is the command to print the jnius classpath:
```python
import jnius_config
print(jnius_config.get_classpath())
```
Add to the code before `import imglyb`

**Comment by hanslovsky** [comment]
@ctrueden I met with @jjahanip and we came to the conclusion that the issues are unrleated. However, with the help of @jjahanip and @jakirkham I was able to resolve that particular build issue on Windows and create a pull request for PyJNIus on conda forge:
https://github.com/conda-forge/staged-recipes/pull/6091
I will also update my anaconda channel tonight with the updated Windows conda package, which should hopefully fix this issue for now.

**Comment by hanslovsky** [comment]
FWIW I had to change all file separators from `/` to `os.sep` to make [imagey](https://github.com/hanslovsky/imagey) work on Windows:
https://github.com/hanslovsky/imagey/commit/cc2e87c96627570c8c86999527ad52034ce25214
You should probably check if the same needs to be done for imagej.py, too.

**Comment by ctrueden** [comment]
Thanks @hanslovsky. I pushed a branch called `file-separators` with UNTESTED changes along those lines. If someone on Windows could verify it works, that would be much appreciated.

**Comment by ctrueden** [comment]
I released imagej 0.3.0 on PyPI. But @jjahanip and @hanslovsky, please note that that release did _not_ include the file separators change, which is still on a topic branch `file-separators`, since no one reported back to me yet with test results. If you tell me that that branch fixes issues for you, I will merge it and cut another release.

**Comment by ctrueden** [comment]
This may be the same issue as #11, now fixed. I released imagej 0.3.1 which I am hoping works on Windows now. Please test.

Closing this issue; please reopen if things are still broken.
